bot/exts/moderation/watchchannels/_watchchannel.py

```diff
@@ -353,8 +353,16 @@ async def prepare_watched_users_data(
             watched_iter = reversed(watched_iter)
 
         list_data["info"] = {}
+
+        # Pre-fetch all watched users to avoid sequential API calls
+        members = await asyncio.gather(*[
+            get_or_fetch_member(ctx.guild, user_id)
+            for user_id, _ in watched_iter
+        ])
+        members = {member.id: member for member in members if member is not None}
+
         for user_id, user_data in watched_iter:
-            member = await get_or_fetch_member(ctx.guild, user_id)
+            member = members.get(user_id)
```

Comment:
This will not contain all `user_id`s from `watched_iter` because members that could not be found will be removed in the line above (`... for member in members if member is not None`). This means that the lines below will fail with a `TypeErro`/`AttributeError` if I understand things correctly.

We need to handle `member` being None here.

Reply:
This is already handled further down.